### PR TITLE
Migrate FSTQueryData to C++ QueryData

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/local/query_data.h [hunk 1]
```diff
@@ -62,30 +65,41 @@ class QueryData {
...
    * time.
    */
-  // TODO(rsgowman): Define once WatchStream::EmptyResumeToken exists.
-  // QueryData(const core::Query& query, int target_id, QueryPurpose purpose);
+  QueryData(core::Query query,
+            int target_id,
+            model::ListenSequenceNumber sequence_number,
+            QueryPurpose purpose);
+
+  /** Creates an invalid QueryData. Prefer QueryData::Invalid() for readability.
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/local/query_data.h]: newline

File: Firestore/core/src/firebase/firestore/remote/remote_event.h [hunk 2]
```diff
@@ -65,10 +64,10 @@ class TargetMetadataProvider {
       model::TargetId target_id) const = 0;
 
   /**
-   * Returns the FSTQueryData for an active target ID or 'null' if this query
-   * has become inactive
+   * Returns the QueryData for an active target ID or 'null' if this query has
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/remote/remote_event.h]: nit: `null` => `nullopt` ?

File: Firestore/core/src/firebase/firestore/local/query_data.h (hunk 1 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/local/query_data.h]: Done.

File: Firestore/core/src/firebase/firestore/remote/remote_event.h (hunk 2 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/remote/remote_event.h]: Done.

**Pull request merged**